### ISSUE TITLE
icyque broken since at least June 2020

### ISSUE THREAD

**Issue by GrandAdmiralThrawn** [EionRobb/icyque#42]
In recent days, icyque appears to be broken on my machine. Receiving messages still works, but when sending a message to any ICQ contact, PidGin would just crash due to a SIGSEGV (signal #11).

Attached you'll find the stderr output of PidGin, as well as its debug log. It crashes just after receiving the Code 200 OK response from the ICQ server. So the messages are actually being sent successfully, but chatting is a bit tedious if you have to restart PidGin after every message sent. ;)

Note that some irrelevant clutter has been removed from the debug log, and several pieces of user-identifiable information have been replaced:

- My sender UIN has been replaced with "<MYICQUIN>"
- My ICQ user name has been replaced with "<MYICQUSERNAME>"
- The receiver's UIN has been replaced with "<TARGETICQUIN>"
- The receiver's ICQ user name has been replaced with "<TARGETICQUSER>"
- My local home directory has been replaced with "<MYPOSIXHOMEDIR>"

I hope I haven't forgotten something.

**Comment by vitalyster** [comment]
That log does not look like icyque output, that is old oscar plugin

**Comment by GrandAdmiralThrawn** [comment]
Huh? If I click on "Account" in PidGin, it shows it as "<UIN> ICQ (WIM)". WIM's supposed to be the new protocol, right?

I also got the duplicate messages that other people have reported in #2 as well.

I guess I'll need to look into my installation, although I don't know what to look at exactly.. I will report back if I find a user error on my end.

**Comment by GrandAdmiralThrawn** [comment]
**Edit:** I looked at my setup, and can't even find libicq.so or liboscar.so anymore. It seems I've deleted them. PidGin loads libicyque.so from ~/.purple/ on my system:

`$ lsof -p <PidGin PID> | grep -i icq` => Returns nothing
`$ lsof -p <PidGin PID> | grep -i oscar` => Returns nothing
`$ lsof -p <PidGin PID> | grep -i icy` => Returns pidgin  2411 MYUSER mem REG 253,0 296105 12988097 /home/MYUSER/.purple/plugins/libicyque.so

When you look at the debug log, the messages are prefixed with "icyque:" just next to the time stamp.

Looks like I'm using icyque to me?

**Comment by vitalyster** [comment]
Nevermind, I misread logs :)

**Comment by EionRobb** [comment]
Sorry for the delay getting back to you, @GrandAdmiralThrawn not sure why I didn't get emailed...

As per the instructions in the stderr.txt, is it possible to follow http://developer.pidgin.im/wiki/GetABacktrace to get a backtrace?

**Comment by GrandAdmiralThrawn** [comment]
> Sorry for the delay getting back to you, @GrandAdmiralThrawn not sure why I didn't get emailed...
> 
> As per the instructions in the stderr.txt, is it possible to follow http://developer.pidgin.im/wiki/GetABacktrace to get a backtrace?

Of course, backtrace file is attached! But this is a release build of Pidgin, not a debug one, so I'm not sure how useful this will be.

File:
[pidgin-backtrace.log](https://github.com/EionRobb/icyque/files/4986151/pidgin-backtrace.log)

I haven't yet found a way to configure my old EL6 system to create core dumps for crashing userspace applications. Will look into it further, if you need a core dump.
